internal/handlers/lokistack_create_or_update.go

```diff
@@ -61,6 +63,13 @@ func CreateOrUpdateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client
 		)
 	}
 
+	sha := sha1.New()
+	sha.Write([]byte(fmt.Sprintf("%v", s3secret)))
+	sha1Storage := fmt.Sprintf("%x", sha.Sum(nil))
```

Comment:
This looks a bit orphaned here. Reading through this function makes me wonder about this low level details. I believe the sha1 should be a extra return value of `secrets.Extract` above.

Reply:
moved to `secrets.Extract`